tests/sip/test_client.py

```diff
@@ -317,6 +317,22 @@ def test_location_code_is_optional(self):
         )
         assert with_code.endswith("COlogin_password|CPlocation_code")
 
+    def test_institution_id_field_is_required(self):
+        without_institution_arg = self.sip.patron_information_request(
+            "patron_identifier", "patron_password"
+        )
+        assert without_institution_arg.startswith('AO|', 31)
+
+    def test_institution_id_field_value_is_optional(self):
```

Comment:
You took `institution_id` out of the signature of this method, so I don't think the code you're testing exists anymore -- that's why you get a test failure. And the institution_id field value is _not_ optional -- it's required, though it can be empty. So I think you can just remove this test.

Reply:
> You took `institution_id` out of the signature of this method, so I don't think the code you're testing exists anymore -- that's why you get a test failure. 

Sorry, I'm not sure what you mean here. The institution_id = '' reference (and = 'MAIN') was a mental error when I wrote the test. I'm thinking that shouldn't be there. 

I see changing the default value of institution_id in the MockSIPClient's __init__ method to match the actual class is needed (I think that's what you were referring to above). And that should satisfy the value setting for the "blank" test. I'm assuming **to test the specific value, I would add an assignment `self.institution_id = 'MAIN'` before the 'with_institution_provided` line?**

And, by naming conventions, I may have muddied the waters a bit. It looks like we need either two or three tests here. For the first, we just need to test the presence of the _field_, it specifier, not its value. In the code, we hard-code the inclusion of the AO field, but we didn't have a test to check that that hard-coding remains intact.

Second, we need to test the field's value (which I called field_value, and used optional to specify that the _field value_ is optional--not the field itself. There's maybe a better name for that so it's not so confusing (on the other hand, having a required field that doesn't require a non-zero/non-blank setting seems odd, but there we are). I didn't quite know what to call the test, but field_value_is_optional seemed right when I wrote it :-) . Maybe we don't need a value test; but it seemed like it would be good to test whether a provided value actually gets transmitted in the message.

I'm thinking my test of a blank value is superfluous now.

How would you handle the test(s)/naming?

Reply:
I think testing an empty-string value is a good idea, because it will verify that the AO field is sent out even when "no value" is provided.